src/test/java/DecoraterTest.java

```diff
@@ -0,0 +1,17 @@
+public class DecoraterTest {
+    public static class CakeShop {
+        public static void main(String[] args) {
```

Comment:
be careful throwing your tests into the main() method - but otherwise great content